libtrip-notification/src/test/java/com/mapbox/navigation/trip/notification/MapboxTripNotificationTest.kt

```diff
@@ -77,6 +84,44 @@ class MapboxTripNotificationTest {
                 any()
             )
         } returns (mockedBroadcastReceiverIntent)
+        every { mockedContext.unregisterReceiver(any()) } just Runs
         return mockedContext
     }
+
+    @Test
+    fun whenTripStartedThenRegisterReceiverCalledOnce() {
+        notification.onTripSessionStarted()
+        verify(exactly = 1) { mockedContext.registerReceiver(any(), any()) }
+    }
+
+    @Test
+    fun whenTripStoppedThenCleanupIsDone() {
+        val notificationManager =
+            mockedContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notification.onTripSessionStopped()
+
+        verify(exactly = 1) { mockedContext.unregisterReceiver(any()) }
+        verify(exactly = 1) { notificationManager.cancel(any()) }
```

Comment:
Those could possibly verify invocations using actual instances instead of `any()` matcher.